connectors/protocol/connectors.py

```diff
@@ -784,11 +784,19 @@ async def prepare(self, config, sources):
             doc["service_type"] = configured_service_type
             self.log_debug(f"Populated service type {configured_service_type}")
 
+        simple_config = source_klass.get_simple_configuration()
+        current_config = self.configuration.to_dict()
+        missing_keys = simple_config.keys() - current_config.keys()
         if self.configuration.is_empty():
             # sets the defaults and the flag to NEEDS_CONFIGURATION
-            doc["configuration"] = source_klass.get_simple_configuration()
+            doc["configuration"] = simple_config
             doc["status"] = Status.NEEDS_CONFIGURATION.value
             self.log_debug("Populated configuration")
+        elif missing_keys:
+            doc["configuration"] = self.updated_configuration(
+                missing_keys, current_config, simple_config
+            )
+            # doc["status"] = Status.NEEDS_CONFIGURATION.value # not setting status, because it may be that default values are sufficient
```

Comment:
I do think we should set it to `NEEDS_CONFIGURATION`, which can prompt the user to update configuration. 

`NEEDS_CONFIGURATION` status can also stop new sync job from being scheduled:

https://github.com/elastic/connectors-python/blob/88a0abbad169dfc5f87ce890759a0835bffbaf5c/connectors/services/job_scheduling.py#L81-L85

Reply:
@danajuratoni, @sphilipse, and I discussed this option, and felt that it would be less interruptive for a customer to not be asked to re-configure. We want migrations to preserve current behavior, so as long as our default values on upgrade keep the same behavior as before the upgrade, there's no need to interrupt any scheduled syncs and force a user to manually configure anything.

Reply:
@seanstory's comment is aligned with my expectations. If there is a possibility to make this non-blocking we should take that